firestore/google/cloud/firestore_v1beta1/_helpers.py

```diff
@@ -119,6 +120,56 @@ def __ne__(self, other):
             return not equality_val
 
 
+class FieldPath(object):
+    """ Field Path object for client use.
+
+    Args:
+        parts (string or iterable): Indicating path of the key to be used.
+    """
+
+    def __init__(self, *parts):
+        invalid_characters = '~*/[]'
+        if len(parts) == 1:
```

Comment:
I don't think there should be special splitting behavior when len(parts) == 1. When I write `FieldPath("a.b")`, I mean a field path of one component which is `a.b`. The place for splitting is in `update`.